File: Python/NeetCode Blind 75/Arrays and Hashing/duplicateInt.py
# Given an integer array nums, return true if any value appears more than once in the array, otherwise return false.

# Example 1
# Input: nums = [1, 2, 3, 3]
# Output: true

# Example 2
# Input: nums = [1, 2, 3, 4]
# Output: false

def hasDuplicate(nums) -> bool:
    # Create dictionary
    # Append to dict if the value does not exist
    # if val does exist then we return true
    # if we reach the end of the nums arr return false

    dict = {}
    
    i = 0
    for num in nums:
        if(dict.get(num) == None):
            dict[num] = num
        else:
            return True
        i += 1
    
    return False

def main():
    # nums = [1,2,3,3]
    nums = [1, 2, 3, 4]
    val = hasDuplicate(nums)
    print(val)

main()

File: C#/NeetCode Blind 150/ArraysAndHashing/duplicateInt.cs
// Given an integer array nums, return true if any value appears more than once in the array, otherwise return false.

// Example 1
// Input: nums = [1, 2, 3, 3]
// Output: true

// Example 2
// Input: nums = [1, 2, 3, 4]
// Output: false

public class Solution {
    public bool hasDuplicate(int[] nums) {
        // Loop through and use a dict to check if value exists
        // if it does return true and if not return False at the end of method

        HashSet<int> set = new HashSet<int>();

        foreach(int num in nums) {
            if(set.Contains(num)) {
                return true;
            }
            set.Add(num);
        }

        return false;
    }
}
